components/ContactForm/ContactForm.tsx

```diff
@@ -0,0 +1,102 @@
+'use client';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import emailjs from '@emailjs/browser';
+
+import { Button } from '@/components/ui/button';
+import {
+	Form,
+	FormItem,
+	FormLabel,
+	FormField,
+	FormControl,
+	FormMessage,
+} from '../ui/form';
+import { Input } from '../ui/input';
+import { Textarea } from '../ui/textarea';
+import { formFields } from '@/constant';
+import { toast } from '../ui/use-toast';
+
+const formSchema = z.object({
+	name: z.string(),
+	email: z.string().email(),
+	messageText: z.string(),
+	senderContact: z.string().email().or(z.string()),
+});
+
+export default function ContactForm() {
+	const form = useForm<z.infer<typeof formSchema>>({
+		resolver: zodResolver(formSchema),
+	});
+
+	const onSubmit = async (data: z.infer<typeof formSchema>) => {
+		const { name, email, messageText, senderContact } = data;
+		try {
+			const templateParams = {
+				to_name: name,
+				to_email: email,
+				to_message: messageText,
+				to_contact: senderContact,
+			};
+			await emailjs.send(
+				process.env.NEXT_PUBLIC_EMAILJS_SERVICE_ID as string,
+				process.env.NEXT_PUBLIC_EMAILJS_TEMPLATE_ID as string,
+				templateParams,
+				process.env.NEXT_PUBLIC_EMAILJS_API_KEY as string
```

Comment:
Nothing wrong with this but:
Instead of casting it as a string, you can also use non-null assertion-
```suggestion
				process.env.NEXT_PUBLIC_EMAILJS_SERVICE_ID!
				process.env.NEXT_PUBLIC_EMAILJS_TEMPLATE_ID!
				templateParams,
				process.env.NEXT_PUBLIC_EMAILJS_API_KEY!
```

Reply:
Most of this PR was UI components from ShadCN, you can feel free to merge this once you review feedback- As I know our active times don't usually match.

Reply:
Thanks @brandonetter